### ISSUE TITLE
Feature gating on NTDDI_VERSION value

### ISSUE THREAD

**Issue by docbrown** [retep998/winapi-rs#371]
The [`NTDDI_VERSION` macro](https://msdn.microsoft.com/en-us/library/windows/desktop/aa383745(v=vs.85).aspx) seems to be the de-facto way to indicate which version of Windows is being targeted (although C/C++ projects are also expected to define `_WIN32_WINNT` for compatibility). Would you consider adding Cargo features for each NTDDI level (`ntddi_win7`, `ntddi_win8`, etc) that control which items are exported?

[`clang-sys`](https://github.com/KyleMayes/clang-sys) does something similar and then automatically defines features like `gte_clang_3_5` for items that are available in Clang 3.5+ and so on.

**Comment by retep998** [comment]
If this was handled by cargo features, the user would have no way to ask to target a lower version of Windows because if any dependency asks for a higher version, the user has no choice but to modify that dependency themselves. Whereas if the option of which Windows version to target was left to the user, the user could decide, and then winapi can inform downstream dependencies which version was selected and they can cfg their code appropriately between using newer functions or using fallback code with older functions. Although downstream dependencies could still check if they are actually running on a newer version of Windows and use `GetProcAddress` to enable newer functionality at runtime, but then they'll need to have access to all the constants and types used by the newer Windows still. That means a feature like this would have to let you use constants and types from any version, with only functions being gated on targeting the newer Windows, but it kinda loses its value as a result. That said for store apps `GetProcAddress` isn't allowed so it could work better there, but Rust doesn't even support targeting store apps yet.

tl;dr It doesn't work very well, so I've gone the easy route of just enabling everything always, and the user is responsible for not calling stuff that requires too new versions of Windows.

**Comment by docbrown** [comment]
That makes sense. Sounds like you've considered this before. Thanks for the quick response!
